.github/workflows/checkdiff.yml

```
@@ -12,6 +14,17 @@ jobs:
           git remote add upstream "${{ github.event.pull_request.base.repo.clone_url }}"
           git fetch upstream
       - name: Checkdiff
+        id: checkdiff
         working-directory: rgbds
         run: |
-          make checkdiff "BASE_REF=${{ github.event.pull_request.base.sha }}" Q= | tee log
+          make checkdiff "BASE_REF=${{ github.event.pull_request.base.sha }}" | tee log
+      - name: Comment
+        uses: actions/github-script@v7
+        with:
+          script: |
+            github.rest.issues.createComment({
+              issue_number: context.issue.number,
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              body: `**Checkdiff:** <pre><code>${{steps.checkdiff.outputs.response}}</code></pre>`
+            })
```

```suggestion
          make checkdiff "BASE_REF=${{ github.event.pull_request.base.sha }}"
      - name: Comment
        if: ${{ failure() && steps.checkdiff.conclusion == 'failure' }} # Only create the review if `checkdiff` found some errors.
        uses: actions/github-script@v7
        with:
          script: |
            github.rest.pulls.createReview({
              pull_number: context.issue.number,
              owner: context.repo.owner,
              repo: context.repo.repo,
              body: `**Checkdiff:** <pre><code>${{ steps.checkdiff.outputs.response }}</code></pre>`,
              event: "REQUEST_CHANGES",
            })
```

...and we'd make `checkdiff` return a non-zero error status if any checks fail?

---

I don't think `checkdiff` should ever fail that way: false positives are expected sometimes, and CI as a whole shouldn't be marked as :x: just because of that.